src/main/java/org/broadinstitute/hellbender/tools/spark/validation/CompareBaseQualities.java

```diff
@@ -0,0 +1,235 @@
+package org.broadinstitute.hellbender.tools.spark.validation;
+
+import com.google.common.collect.Lists;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function2;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.programgroups.TestSparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.QualityUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import scala.Tuple2;
+
+import java.io.Serializable;
+import java.util.List;
+
+@CommandLineProgramProperties(summary = "Compares two the quality scores of BAMs", oneLineSummary = "Diff qs of two BAMs",
+        programGroup = TestSparkProgramGroup.class)
+
+public class CompareBaseQualities extends GATKSparkTool  {
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public boolean requiresReads() { return true; }
+
+    @Argument(doc="the second BAM", shortName = "I2", fullName = "input2", optional = false)
+    protected String input2;
+
+    @Argument(doc="print matrix", shortName = "pm", fullName = "printMatrix", optional = true)
+    protected Boolean printMatrix = false;
+
+    @Argument(doc="print summary", shortName = "ps", fullName = "printSummary", optional = true)
+    protected Boolean printSummary = true;
+
+    @Argument(doc="crash on diff", shortName = "cd", fullName = "crashOnDiff", optional = true)
+    protected Boolean crashOnDiff = false;
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+        JavaRDD<GATKRead> firstReads = getReads();
+        ReadsSparkSource readsSource2 = new ReadsSparkSource(ctx);
+        JavaRDD<GATKRead> secondReads = readsSource2.getParallelReads(input2, getIntervals());
+
+        long firstBamSize = firstReads.count();
+        long secondBamSize = secondReads.count();
+        if (firstBamSize != secondBamSize) {
```

Comment:
Added a comment, we want to "crash on diff" when the user uses the tool correctly, but there are subtle diffs. If the user passes two random BAMs that don't even have the same number of reads, I think it's best to stop here. Otherwise, we keep going and give results hinting that the issue is more subtle.